pkg/cloud/baremetal/actuators/machine/actuator_test.go

```diff
@@ -1947,6 +1947,20 @@ func TestRemediation(t *testing.T) {
 		t.Fail()
 	}
 
+	// Get the remediation power off annotation and check that it matches RebootModeHard
+	remediationString := host.Annotations[requestPowerOffAnnotation]
+	remediationAnnotation := bmh.RebootAnnotationArguments{}
+	err = json.Unmarshal([]byte(remediationString), &remediationAnnotation)
+	if err == nil {
```

Comment:
The `github.com/stretchr/testify/assert` package (already in use elsewhere in this test suite) provides functions for doing these checks. I think this if/else block could be replaced with

```go
assert.Equal(t, bah.RebootModeHard, remediationAnnotation.Mode)
assert.NoError(t, err)
```

Reply:
Yep, you're absolutely right, assert is much cleaner. I'll update. Thanks.